File: 6-kyu/Triangle Type/triangleType.js
// Kata can be found at: https://www.codewars.com/kata/53907ac3cd51b69f790006c5

/* Should return ᐃ type:
  0 : if ᐃ cannot be made with given sides
  1 : acute ᐃ
  2 : right ᐃ
  3 : obtuse ᐃ
*/
function triangleType(a, b, c){



  if ( a > c && a > b)   {
  
    temp = c;
    c = a;
    a = temp;
  
  }  else if(b > c && a < b)  {
  
    temp = c;
    c = b;
    b = temp;
  
  }  
  if ( a == 0 || b == 0 || c == 0)  return 0;
  if (c >= a + b) return 0;
  if (c * c == a * a + b * b)  return 2;
  return (c * c < a * a + b * b) ? 1 : 3;

}

File: 6-kyu/Triangle Type/triangleType.py
# Kata can be found at: https://www.codewars.com/kata/53907ac3cd51b69f790006c5

def triangle_type(a, b, c):
  
  if a > b and a > c:
      temp = c
      c = a
      a = temp
  if b > c and b > a:
      temp = c
      c = b
      b = temp
      
  if not a + b > c or not a + c > b or not b + c > a:
      return 0
  if (c * c == a * a + b * b) and (c * c > 0):
      return 2
  if (c * c < a * a + b * b) and c * c > 0:
      return 1
  if (c * c > a * a + b * b) and (a * a + b * b > 0):
      return 3
  else:
      return 0
